libs/designsystem/src/lib/components/modal/services/modal-navigation.service.ts

```diff
@@ -0,0 +1,197 @@
+  }
+
+  private removeChildSegments(currentRoutePaths: string[], routes: Routes) {
+    if (!currentRoutePaths || !currentRoutePaths.length) return;
+
+    const moduleRelativePaths = this.getRoutePaths(routes, ['']);
+    moduleRelativePaths.sort().reverse(); // Ensure child paths are evaluated first
+    const currentUrl = currentRoutePaths.join('/');
+    let matchedChildRoute = moduleRelativePaths.find((path) => currentUrl.endsWith(path));
+    if (matchedChildRoute) {
+      const relativeChildRoute = matchedChildRoute.startsWith('/')
+        ? matchedChildRoute.substring(1)
+        : matchedChildRoute;
+      const childSegmentCount = relativeChildRoute.split('/').length;
+      currentRoutePaths.splice(-childSegmentCount); // Pop child segments from end of current route path
+    }
+  }
+
+  private getRoutePaths(routes: Routes, parentPath: string[]): string[] {
+    return Array.isArray(routes)
+      ? ([] as string[]).concat(...routes.map((route) => this.getRoutePath(route, parentPath)))
+      : [];
+  }
+
+  private getRoutePath(route: Route, parentPath: string[]): string[] {
+    const routes: string[] = [];
+    if (!!route.outlet) return routes; // Don't return relative paths for outlet routes
+
+    const currentPath = [...parentPath];
+    if (!!route.path) {
+      currentPath.push(route.path);
+      routes.push(currentPath.join('/'));
+    }
+
+    return routes.concat(this.getRoutePaths(route.children, currentPath));
+  }
+
+  private getModalRoutePath(route: Route, parentPath: string[]): string[] {
+    const modalOutletName = 'modal';
+    if (!!route.path && route.outlet === modalOutletName) {
+      const modalOutletPath = `(${modalOutletName}:${route.path})`;
+      const modalRoutePath = [...parentPath, modalOutletPath].join('/');
+      return [modalRoutePath];
+    }
+    const currentPath = [...parentPath];
+    if (!!route.path) {
+      currentPath.push(route.path);
+    }
+    return ([] as string[]).concat(...this.getModalRoutePaths(route.children, currentPath));
+  }
+
+  private getModalRoutePaths(routes: Routes, parentPath: string[]): string[] {
+    return Array.isArray(routes)
+      ? ([] as string[]).concat(...routes.map((route) => this.getModalRoutePath(route, parentPath)))
+      : [];
+  }
+
+  private isNewModalWindow(navigationEnd: NavigationEnd) {
+    const currentNavigation = this.router.getCurrentNavigation();
+    if (!currentNavigation || !currentNavigation.previousNavigation) {
+      return true;
+    }
+    const previousRoutePath = (
+      currentNavigation.previousNavigation.finalUrl ||
+      currentNavigation.previousNavigation.extractedUrl
+    ).toString();
+    const wasModalRoute = this.isModalRoute(previousRoutePath);
+    const isModalRoute = this.isModalRoute(navigationEnd.urlAfterRedirects);
+    if (!wasModalRoute && isModalRoute) {
+      return true;
+    }
+
+    const currentModalRouteParent = navigationEnd.urlAfterRedirects.split('/(modal:')[0];
```

Comment:
Consider extracting magic string to a prop since it's used multiple places

Reply:
Fixed 👍